src/pages/filter/ui/modal-edit/components/modal-inheritance-mode/components/inheritance-mode-content.tsx

```diff
@@ -27,11 +28,23 @@ export const InheritanceModeContent = observer(
     const handleCheckGroupItem = (checked: boolean, name: string) => {
       if (checked) {
         dtreeStore.addSelectedFilter(name)
-      } else {
-        dtreeStore.removeSelectedFilter(name)
+        return
       }
+
+      dtreeStore.removeSelectedFilter(name)
+    }
+
+    const setAll = (checked: boolean) => () => {
+      const allVariants = toJS(variants)
+      allVariants &&
+        allVariants.forEach((variant: any[]) =>
+          handleCheckGroupItem(checked, variant[0]),
+        )
     }
 
+    const clearAll = setAll(false)
+    const selectAll = setAll(true)
```

Comment:
1. Move this logic to ModalInheritanceModeStore
2. Don't use currying where it's not needed, keep the code as obvious as possible